### PR TITLE
add 'executorService.shutdown()' call to 'Server.stop()'

### PR DESCRIPTION
**Pull request by eemhu**
Fix for issue #59

### CODE REVIEW

**Comment by kortemik** [comment]
this is actually already present in SocketPoll.close()

**Comment by kortemik** [comment]
i think the appropriate place would be after 
```
            while (!stop.get()) {
                socketPoll.poll();
            }
```
because socketPoll use the factory and there might be an active poll in progress and therefore it wouldn't close cleanly.

File: src/main/java/com/teragrep/rlp_03/Server.java [hunk 1]
```diff
@@ -115,6 +115,8 @@ public void run() {
             while (!stop.get()) {
                 socketPoll.poll();
             }
+            // shutdown executorService when outside of poll() loop
+            executorService.shutdown();
```
Comment (kortemik) [inline src/main/java/com/teragrep/rlp_03/Server.java:119]: actually when rethinking the finally {} might be even more appropriate so the executorService would be closed even if socketPoll throws

**Review by kortemik — CHANGES_REQUESTED** [review]

**Review by kortemik — APPROVED** [review]

**Comment by kortemik** [comment]
needs rebase

**Comment by eemhu** [comment]
rebased on newest main branch.

**Pull request merged**